doc/neps/nep-0056-array-api-main-namespace.rst

```diff
@@ -0,0 +1,524 @@
+have shown that the separate strict implementation and separate array object
+are mostly good for testing purposes, but not for regular usage in downstream
+libraries. Having support in the main namespace resolves this issue. Hence this
+NEP supersedes NEP 47. The ``numpy.array_api`` module will be moved to a
+standalone package, to facilitate easier updates not tied to a NumPy release
+cycle.
+
+Some of the key design rules from the array API standard (e.g., output dtypes
+predictable from input dtypes, no polymorphic APIs with varying number of
+returns controlled by keywords) will also be applied to NumPy functions that
+are not part of the array API standard, because those design rules are now
+understood to be good practice in general. Those two design rules in particular
+make it easier for Numba and other JIT compilers to support NumPy or
+NumPy-compatible APIs. We'll note that making existing arguments
+positional-only and keyword-only is a good idea for functions added to NumPy in
+the future, but will not be done for existing functions since each such change
+is a backwards compatibility break and it's not necessary for writing code that
+is portable across libraries supporting the standard. An additional reason to
+apply those design rules to all functions in the main namespace now is that it
+then becomes much easier to deal with potential standardization of new
+functions already present in NumPy - those could otherwise be blocked or forced
+to use alternative function names due to the need for backwards compatibility.
+
+It is important that new functions added to the main namespace integrate well
+with the rest of NumPy. So they should for example follow broadcasting and
+other rules as expected, and work with all NumPy's dtypes rather than only the
+ones in the standard. The same goes for backwards-incompatible changes (e.g.,
+linear algebra functions need to all support batching in the same way, and
+consider the last two axes as matrices). As a result, NumPy should become more
+rather than less consistent.
+
+We'll note that one remaining incompatibility will be that NumPy is returning
+array scalars rather than 0-D arrays in most cases where the standard, and
+other array libraries, return 0-D arrays (e.g., indexing and reductions). There
+have been multiple discussions over the past year about the feasibility of
+removing array scalars from NumPy, or at least no longer returning them by
+default. However, this would be a large effort with some uncertainty about
+technical risks and impact of the change, and no one has taken it on. Given
+that array scalars implement a mostly array-compatible interface, this doesn't
+seem like the highest-prio item regarding array API standard compatibility.
+
+Here are what we see as the main expected benefits and costs of the complete
+set of proposed changes:
+
+Benefits:
+
+- It will remove the "having to make a choice between the NumPy API and the
+  Array API" issue for other libraries,
```

Comment:
For consistency
```suggestion
- It will remove the "having to make a choice between the NumPy API and the
  array API" issue for other libraries,
```